src/ConstructiveSolidGeometry/SurfacePrimitives/ConeMantle.jl

```diff
@@ -27,6 +27,55 @@ T: Type of values, e.g. Float64
     rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
 end
 
+radius_at_z(cm::ConeMantle{T,T}, z::T) where {T} = cm.r
+function radius_at_z(cm::ConeMantle{T,Tuple{T,T}}, z::T) where {T} 
+    rslope = (cm.r[2] - cm.r[1]) / 2cm.hZ 
+    return cm.r[1] + rslope * (z + cm.hZ)
+end
+
+"""
+    intersection(cm::ConeMantle{T,Tuple{T,T}}, l::Line{T}) where {T}
+
+The function will always return 2 CartesianPoint's.
+If the line just touches the mantle, the two points will be the same. 
+If the line does not touch the mantle at all, the two points will have NaN's as there coordinates.
+"""
```

Comment:
What if it only crosses once (which is possible)? One point with NaNs and one Real? Please include this for completeness

Reply:
Also what is this used for? I don't find the function being used anywhere

Reply:
Ah, right. Thanks :) I added that case: https://github.com/JuliaPhysics/SolidStateDetectors.jl/pull/163/commits/9b2a22b0b40a68c5bc759196b548199570446af9

Reply:
> Also what is this used for? I don't find the function being used anywhere

Basically in `paint!` -> `evaluate`